lms/djangoapps/discussion/django_comment_client/utils.py

```diff
@@ -146,6 +147,40 @@ def is_user_community_ta(user, course_id):
     return has_forum_access(user, course_id, FORUM_ROLE_COMMUNITY_TA)
 
 
+def get_users_with_roles(roles, course_id):
+    """
+    Get all users with specified roles for a course
+    """
+    users_with_roles = [
+        user
+        for role in Role.objects.filter(
+            name__in=roles,
+            course_id=course_id
+        )
+        for user in role.users.all()
+    ]
+    return users_with_roles
+
+
+def get_users_with_moderator_roles(context):
+    """
+    Get all users within the course with moderator roles
+    """
+    moderators = get_users_with_roles([FORUM_ROLE_ADMINISTRATOR, FORUM_ROLE_MODERATOR,
+                                       FORUM_ROLE_COMMUNITY_TA], context['course_id'])
+
+    context_thread = cc.Thread.find(context['thread_id'])
+    if getattr(context_thread, 'group_id', None) is not None:
+        group_moderators = get_users_with_roles([FORUM_ROLE_GROUP_MODERATOR], context['course_id'])
+        course_discussion_settings = CourseDiscussionSettings.get(context['course_id'])
+        moderators_in_group = [user for user in group_moderators if get_group_id_for_user(
+            user, course_discussion_settings) == context_thread.group_id]
+        moderators += moderators_in_group
+
+    moderators = set(moderators)
```

Comment:
Why set here? Is there a possibility of duplication here?

Reply:
A user can have multiple discussion roles for a course. 
`set` is used here to get rid of the same user added to the moderators list when the user has multiple roles. This will prevent the particular user from getting multiple emails!

Reply:
Thanks for the clarification